packages/lake/src/components/SegmentedControl.tsx

```diff
@@ -34,171 +63,57 @@ export const SegmentedControl = <T extends string>({
   items,
   onValueChange,
 }: Props<T>) => {
-  const indicatorWidth = useRef(0);
-  const itemWidths = useRef<number[]>([]);
-  const [itemsToDisplay, setItemsToDisplay] = useState(items);
   const selectedItemIndex = items.findIndex(item => item.id === selected);
 
-  const maybeHideItems = () => {
-    const maxWidth = itemsToDisplay.length * indicatorWidth.current;
-    const slots = Math.floor(maxWidth / Math.max(...itemWidths.current));
-    if (slots !== itemsToDisplay.length) {
-      setItemsToDisplay(items.slice(0, Math.max(1, slots)));
-    }
-  };
-
   return (
-    <Box
-      direction="row"
-      style={{
-        backgroundColor: colors.gray[50],
-        height: HEIGHT,
-        borderRadius: radii[8],
-      }}
-    >
-      <Box
-        role="none"
-        aria-hidden={true}
-        direction="row"
-        style={{
-          position: "absolute",
-          visibility: "hidden",
-        }}
-      >
-        {itemsToDisplay.map((item, index) => {
-          return (
-            <Box
-              key={`${item.id}-hidden`}
-              direction="row"
-              alignItems="center"
-              onLayout={({ nativeEvent: { layout } }) => {
-                itemWidths.current[index] = layout.width;
-              }}
-              style={{
-                padding: spacings[12],
-                backgroundColor: "green",
-              }}
-            >
-              {isNotNullish(item.icon) && (
-                <>
-                  {item.icon}
-
-                  <Space width={12} />
-                </>
-              )}
-
-              <LakeText
-                color={colors.gray[900]}
-                numberOfLines={1}
-                style={{
-                  userSelect: "none",
-                }}
-              >
-                {item.name}
-              </LakeText>
-            </Box>
-          );
-        })}
-      </Box>
-
-      <View
-        style={{
-          flexGrow: 1,
-          flexShrink: 1,
-          padding: spacings[4],
-        }}
-      >
-        <Box direction="row" style={commonStyles.fill}>
-          <View
-            role="none"
-            style={{
-              position: "absolute",
-              left: 0,
-              top: 0,
+    <Box style={styles.container}>
+      <Box direction={mode === "mobile" ? "column" : "row"}>
+        <View
+          role="none"
+          style={[
+            styles.selectedItem,
+            // eslint-disable-next-line react-native/no-inline-styles
```

Comment:
You can remove the `// eslint-disable-next-line react-native/no-inline-styles`, just need to declare two styles for `{ right: 0 }` and `{ bottom: 0 }`